src/coreclr/src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -133,7 +133,7 @@ public static void AddMemoryPressure(long bytesAllocated)
                         SR.ArgumentOutOfRange_NeedPosNum);
             }
 
-            if ((4 == IntPtr.Size) && (bytesAllocated > int.MaxValue))
+            if ((IntPtr.Size == 4) && (bytesAllocated > int.MaxValue))
```

Comment:
`IntPtr.Size == sizeof(int)`?

Reply:
Isn't this a 32bit check? Don't we have a utility function to return if the system is 32bit? msbuild does something similar:
https://github.com/dotnet/msbuild/blob/master/src/Shared/EnvironmentUtilities.cs#L11

Reply:
No, it is a bitness check for 32bit.
https://github.com/dotnet/runtime/blob/ea2b09beea535abf960651c59359e40f99f2fecf/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L138

Reply:
.... I'm a little surprised we don't have a 32-bit process check...  But it's probably from the changeover 32->64, so code was going to be written as `if(Is64BitProcess) {...}`...

Yes, I'm aware it's a 32-bit check; the use of `sizeof` was just an attempt to avoid a magic number.

Reply:
I think using `sizeof` would only run it when the process bitness and the os bitness match. Which I don't think is the intention of the code.

Reply:
This PR is changing ~600 files.  Please restrict the change to be purely switching the sides of the expression and nothing else.  It's way too likely mistakes will otherwise creep in.

Reply:
You might also consider breaking it into pieces -- doing a few folders in one PR, merging that, and doing another PR, and enabling the rule in the last one. Or just doing 20 typical changes in one PR. That would also help us decide whether we're making the right change.

Reply:
I'd like to do one big change so we see how much is affected. Once we know how much of the code-base is touched then I can split it up with a PR per library or something.

At the moment I am having issues refactoring due to a nuget issue related to I think #32205.

Reply:
`sizeof(<primitive>)` returns a constant value at compile time.  Unlike C++, primitives have a fixed size in C# (which is why there's a request for `nint` - Native INTeger).